### ISSUE TITLE
Workspaces fail to start in cases where certain firewalls are active

### ISSUE THREAD

**Issue by eivantsov** [codenvy/codenvy#1053]
Since swarm container just needs an internal IP of a socat container but the same IP will be used to represent servers and preview URLs, there's a need of having 2 IPs - internal:external.

It should be possible to provide just one though.

This way we won't expose Docker API since swarm will be using an internal IP, while users will have external IPs in the IDE.

**Comment by garagatyi** [comment]
We investigated this with @riuvshin and @eivantsov. Looks like configuring it with a code is not good solution because it either blocks ability or makes i over-complicated to dynamically add/remove docker nodes to swarm cluster.
In all-in-one usecase we should configure existing properties to achieve what we need. In case of scalable AIO or multi-node installation admin will have to setup infrastructure to achieve needed level of security.

@eivantsov can we close this issue?

**Comment by TylerJewell** [comment]
@garagatyi - Can you walk through what the scalable AIO configuration would be for admins with a firewall situation?  It may be ok for us to not change our code, but we should not close this issue until we discuss:
1. Is there any way for us to detect the firewall configuration to alert installer?
2. Is there a way for an admin to test for firewall?
3. is there special steps an admin takes to intsalling with firewall?

If we can have very good docs on this, then we can close next.

**Comment by garagatyi** [comment]
1. I don't know if it is possible since there are a lot of different configurations of firewalls.
2. I think that admin knows which firewall is enabled in a system under his control. I don't know how to test that. See reason above.
3. Again it depends on firewall and infrastructure.

**Comment by garagatyi** [comment]
@eivantsov @riuvshin Do you guys know anything helpful related to the topic?

**Comment by riuvshin** [comment]
@garagatyi no, sorry :(

**Comment by garagatyi** [comment]
I'm closing this issue. Feel free to reopen if you think that something else could be done

**Comment by garagatyi** [comment]
Clarification of impact of firewall to codenvy in container functioning:
1. If Ubuntu firewall with default settings (deny everything) is enabled codenvy stuck at **Server booting...** and neither goes further nor print any message. Although product is OK it is not accessible from browser.
2. If port 80 is allowed only Codenvy starts, but workspaces can't be started
![selection_001](https://cloud.githubusercontent.com/assets/1851863/20563362/f33c684c-b190-11e6-9639-8e570ff214f9.png)
3. If 80, and ephemeral ports are opened behavior the same as in # 2
4. If 80, ephemeral, 23750 ports are opened everything is ok.

We need 80/443 for the product. ephemeral ports for workspaces and 23750 as a docker port of a docker node.


Probably we can improve checks of CLI to find out if required ports are not blocked by firewall and print error message to open these ports if needed. Maybe it is only possible with some script that should be started from the host.

**Comment by garagatyi** [comment]
@TylerJewell can I close this issue?

**Comment by TylerJewell** [comment]
No.  It is not closable until there is clear tests + docs + product capabilities to prevent these issues.

**Comment by garagatyi** [comment]
This issue didn't state DoD as all this stuff. Shall we open new issue with explanation on what exactly should be done instead of continuing working on that issue? @TylerJewell @skabashnyuk

**Comment by TylerJewell** [comment]
No, this has a history in it.  Just let it be.  It's not closable.
